qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/UserAccountSteps.java

```diff
@@ -139,11 +139,44 @@ public void enterDoctorDetails() {
 	@And("a user clicks on save account button")
 	public void clickOnSaveUserAccountButton() {
 		userAccountPage.saveUserAccount();
-		userAccountPage.waitForPage();
```

Comment:
i removed that page waiter because it was failing the test aswell as it hangs for a long time then it breaks. but when i removed it, it can now move to the next step but its also failing

Reply:
but when i use `assertFalse(textExists("Validation errors found Failed to save account details"));` it passes

Reply:
My understanding is that password validation is done at the time when a user is entering the password in the password filed if the password does not meet the criteria then validation error message is displayed.

Reply:
> My understanding is that password validation is done at the time when a user is entering the password in the password filed if the password does not meet the criteria then validation error message is displayed.

that is true for cases of password length and the password field not being left empty. I dont know if thats a bug or something but when i interract with the qaserver, those errors dont happen instantly as you type (like it should be)

Reply:
i have tested that too with the demo server

Reply:
> I dont know if thats a bug or something but when i interract with the qaserver, those errors dont happen instantly as you type (like it should be)

Try to type in a password "pass" in the password field then the validation error message should appear!

Reply:
yes the validation error `At least 8 character(s) are required` appears. challenge is on the other validation errors that dont show. they were all combined in `Validation errors found Failed to save account details` which only appears when the form is submitted

Reply:
As mentioned we can assert that the text_At least 8 character(s) are required_ does not appear when a user type in a qualified password so errors are caught at their respective fields.

Reply:
> the other validation error

i mean for example `passwords must contain at least one non-digit`, then `asswords must contain at least one digit`

Reply:
> As mentioned we can assert that the text_At least 8 character(s) are required_ does not appear when a user type in a qualified password so errors are caught at their respective fields.

alright thanks however i believe for that matter, i would have to structure my feature file like https://github.com/mherman22/openmrs-contrib-qaframework/blob/4ca5d84aad3d8da64a281ef6aa7488baa0a07f7f/qaframework-bdd-tests/src/test/resources/features/refapp-2.x/06-user/users.feature so as to cater for all those errors especially `At least 8 character(s) are required`

Reply:
> then `passwords must contain at least one digit`

That together with `passwords must contain at least Upper case letter` can not be validated within the password field. These two cases can only be caught during submitting of the form and the error thrown is generic(`Validation errors found Failed to save account details`) which does not really tell what the user has to do or where things have been messed up.

@mherman22 @SharonVarghese so what do you think can be tested out realistically?

Reply:
i think, in addition to the [above comment](https://github.com/openmrs/openmrs-contrib-qaframework/pull/388#discussion_r831831287), all those should be scenarios but all causing the same error. for example it should be pointed out that all these validations (if not met) can cause the `Validation errors found Failed to save account details` and this can be achieved by making each of the following a scenario on its own.

1. `passwords must contain at least Upper case letter`,
2. `passwords must contain at least one digit`, 
3. `At least 8 character(s) are required`

Take a look at what i had done initially ---> https://github.com/mherman22/openmrs-contrib-qaframework/blob/4ca5d84aad3d8da64a281ef6aa7488baa0a07f7f/qaframework-bdd-tests/src/test/resources/features/refapp-2.x/06-user/users.feature

Reply:
@mherman22 your original test design is fair enough, we can resort back to that design.

Reply:
Well done @mherman22 , try to check through the line 19 of this class, Seems your assertions have issues.